src/pepr/operator/reconcilers/package-reconciler.ts

```diff
@@ -54,10 +55,12 @@ export async function packageReconciler(pkg: UDSPackage) {
 
     // Configure SSO
     const ssoClients = await keycloak(pkg);
+    const authserviceClients = await authservice(pkg);
```

Comment:
There is a potential security issue where this can fail open:

1. Create a Package without groups, and wait for UDS operator to create the VirtualService
2. Update the Package to have authservice (and optionally groups). If the operator has an error in these SSO functions, these will throw an error and halt the reconcile... but the old VirtualService will stay around. This means that it fails open.

(similarly you could go from Package+authservice → Package+authservice+groups and get an error in group application and get similar bad fail-open)

I'm unsure what the best solution is. Should we maybe catch errors in the authservice code and tear down the virtual services? That would mean the operator fails closed? Perhaps move `istioResources()` below this code and pass it a bool `destroy` option that triggers teardown (this allows better code reuse)?

Reply:
:+1: Created an issue as a follow up here - https://github.com/defenseunicorns/uds-core/issues/519